bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/api/AzureActions.java

```diff
@@ -0,0 +1,122 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+/**
+ * The {@link AzureActions} class contains AWS S3 API implementation.
+ *
+ * @author Alexandr Salamatov - Initial contribution
+ */
+@NonNullByDefault
+public class AzureActions {
+    private final HttpClient httpClient;
+    private static final Duration REQUEST_TIMEOUT = Duration.ofMinutes(1);
+    private static final String CONTENT_TYPE = "application/xml";
+    private URL containerUri;
+
+    public AzureActions(HttpClientFactory httpClientFactory, String accountName, String containerName) {
+        this.httpClient = httpClientFactory.getCommonHttpClient();
+        try {
+            this.containerUri = new URL("http://" + accountName + ".blob.core.windows.net/" + containerName);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException("Unable to parse service endpoint: " + e.getMessage());
+        }
+    }
+
+    public List<String> listContainer(String prefix) throws Exception {
+        Map<String, String> headers = new HashMap<String, String>();
+        Map<String, String> params = new HashMap<String, String>();
+        return listBlob(prefix, headers, params);
+    }
+
+    public List<String> listBlob(String prefix, Map<String, String> headers, Map<String, String> params)
+            throws Exception {
+
+        params.put("restype", "container");
+        params.put("comp", "list");
+        params.put("maxresults", "1000");
+        params.put("prefix", prefix);
+        headers.put(ACCEPT.toString(), CONTENT_TYPE);
+        // headers.put("x-ms-version", "2020-04-08");
+
+        Request request = httpClient.newRequest(this.containerUri.toString()) //
+                .method(GET) //
+                .timeout(REQUEST_TIMEOUT.toNanos(), TimeUnit.NANOSECONDS); //
+
+        for (String headerKey : headers.keySet()) {
+            request.header(headerKey, headers.get(headerKey));
+        }
+        for (String paramKey : params.keySet()) {
+            request.param(paramKey, params.get(paramKey));
+        }
+
+        ContentResponse contentResponse = request.send();
+        if (contentResponse.getStatus() != 200) {
+            throw new Exception("HTTP Response is not 200");
+        }
+
+        DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder docBuilder = docBuilderFactory.newDocumentBuilder();
+        // This returns extra character before <xml. Need to find out why
+        String sResponse = contentResponse.getContentAsString();
+        InputSource is = new InputSource(new StringReader(sResponse.substring(sResponse.indexOf("<"))));
```

Comment:
I'm doing this, but I don't like it.
The reason why I'm doing this that Azure in the HTTP response has some additional character before <?xml statement. So the DOM parser fails.

![2023-05-02 16 24 29](https://user-images.githubusercontent.com/40123561/235789579-560c2788-6d5a-4791-8ac1-65ea012f024d.jpg)

Did anybody experience that and if yes, what is the best solution?

Reply:
@wborn did you experience similar issue with Azure?

Reply:
Not really, maybe it's a BOM?

https://en.m.wikipedia.org/wiki/Byte_order_mark